mypy/semanal.py

```diff
@@ -2366,11 +2374,19 @@ def fail_typeddict_arg(self, message: str,
 
     def build_typeddict_typeinfo(self, name: str, items: List[str],
                                  types: List[Type]) -> TypeInfo:
-        mapping_value_type = join.join_type_list(types)
         fallback = (self.named_type_or_none('typing.Mapping',
```

Comment:
Probably the most precise fallback would be just ``Dict``, but probably we want to have some flexibility because ``Mapping`` is covariant in value type (unlike ``MutableMapping`` and ``Dict`` that are invariant).